src/encoder.rs

```diff
@@ -27,20 +29,34 @@ pub enum Error {
     IoError(StdIoError),
 }
 
-impl fmt::Debug for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+impl Error {
+    #[doc(hidden)]
+    fn __description(&self) -> &str {
+        use std::error::Error;
         match *self {
-            NestedOptions => "nested Option types are not supported".fmt(f),
-            UnsupportedType => "unsupported type".fmt(f),
-            InvalidStr => "invalid str".fmt(f),
-            MissingElements => "no elements in value".fmt(f),
-            KeyIsNotString => "key is not a string".fmt(f),
-            NoFilename => "a filename must be provided".fmt(f),
-            IoError(ref err) => err.fmt(f),
+            NestedOptions => "nested Option types are not supported",
+            UnsupportedType => "unsupported type",
+            InvalidStr => "invalid str",
+            MissingElements => "no elements in value",
+            KeyIsNotString => "key is not a string",
+            NoFilename => "a filename must be provided",
+            IoError(ref err) => err.description(),
         }
     }
 }
 
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        self.__description().fmt(f)
+    }
+}
+
+impl error::Error for Error {
+    fn description(&self) -> &str {
+        self.__description()
```

Comment:
This is how Display and Error are implemented in std for a few types. I tried putting the impl in Display first, that won't work, but it might work the other way. I'll try it.